CMakeLists.txt

```diff
@@ -24,213 +14,559 @@ set(OC_IDD_API_ENABLED ON CACHE BOOL "Enable the Introspection Device Data API."
-    ${PROJECT_SOURCE_DIR}/api/oc_discovery.c
-    ${PROJECT_SOURCE_DIR}/api/oc_endpoint.c
-    ${PROJECT_SOURCE_DIR}/api/oc_enums.c
-    ${PROJECT_SOURCE_DIR}/api/oc_helpers.c
-    ${PROJECT_SOURCE_DIR}/api/oc_main.c
-    ${PROJECT_SOURCE_DIR}/api/oc_mnt.c
-    ${PROJECT_SOURCE_DIR}/api/oc_network_events.c
-    ${PROJECT_SOURCE_DIR}/api/oc_rep.c
-    ${PROJECT_SOURCE_DIR}/api/oc_resource_factory.c
-    ${PROJECT_SOURCE_DIR}/api/oc_ri.c
-    ${PROJECT_SOURCE_DIR}/api/oc_server_api.c
-    ${PROJECT_SOURCE_DIR}/api/oc_session_events.c
-    ${PROJECT_SOURCE_DIR}/api/oc_swupdate.c
-    ${PROJECT_SOURCE_DIR}/api/oc_uuid.c
-    ${PROJECT_SOURCE_DIR}/api/c-timestamp/timestamp_compare.c
+if(OC_OSCORE_ENABLED)
+    list(APPEND PUBLIC_COMPILE_DEFINITIONS "OC_OSCORE")
+endif()
+
+if(OC_IPV4_ENABLED)
+    list(APPEND PUBLIC_COMPILE_DEFINITIONS "OC_IPV4")
+endif()
+
+if(OC_DNS_LOOKUP_IPV6_ENABLED)
+    list(APPEND PRIVATE_COMPILE_DEFINITIONS "OC_DNS_LOOKUP_IPV6")
+endif()
+
+######## Gather source files ########
+file(GLOB COMMON_SRC
     ${PROJECT_SOURCE_DIR}/api/c-timestamp/timestamp_format.c
-    ${PROJECT_SOURCE_DIR}/api/c-timestamp/timestamp_parse.c
-    ${PROJECT_SOURCE_DIR}/api/c-timestamp/timestamp_tm.c
     ${PROJECT_SOURCE_DIR}/api/c-timestamp/timestamp_valid.c
+    ${PROJECT_SOURCE_DIR}/api/c-timestamp/timestamp_parse.c
+    ${PROJECT_SOURCE_DIR}/util/*.c
 )
 
-if(OC_CLOUD_ENABLED)
-    target_sources(iotivity-api INTERFACE
-        ${PROJECT_SOURCE_DIR}/api/cloud/oc_cloud_manager.c
-        ${PROJECT_SOURCE_DIR}/api/cloud/oc_cloud_rd.c
-        ${PROJECT_SOURCE_DIR}/api/cloud/oc_cloud_resource.c
-        ${PROJECT_SOURCE_DIR}/api/cloud/oc_cloud_store.c
-        ${PROJECT_SOURCE_DIR}/api/cloud/oc_cloud.c
-        ${PROJECT_SOURCE_DIR}/api/cloud/rd_client.c
-        ${PROJECT_SOURCE_DIR}/api/cloud/oc_cloud_apis.c
+if(OC_PKI_ENABLED)
+    list(APPEND COMMON_SRC
+        ${PROJECT_SOURCE_DIR}/api/c-timestamp/timestamp_tm.c
     )
 endif()
 
-target_include_directories(iotivity-api INTERFACE
-    ${PROJECT_SOURCE_DIR}/
-    ${PROJECT_SOURCE_DIR}/api
-    ${PROJECT_SOURCE_DIR}/api/c-timestamp
-    ${PROJECT_SOURCE_DIR}/include
+file(GLOB SERVER_SRC
+    ${PROJECT_SOURCE_DIR}/messaging/coap/*.c
+    ${PROJECT_SOURCE_DIR}/api/*.c
 )
 
-# Core functions used by Iotivity
-add_library(iotivity-common INTERFACE)
+set(CLIENT_SRC ${SERVER_SRC})
+
+# Detect the platform and pick the right port
+if(UNIX)
+    file(GLOB PORT_SRC port/linux/*.c)
+    set(PORT_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/port/linux)
+elseif(WIN32)
+    file(GLOB PORT_SRC port/windows/*.c)
+    set(PORT_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/port/windows)
+else()
+    message(ERROR "Can build only on Linux and Windows!")
+endif()
+list(APPEND SERVER_SRC ${PORT_SRC})
+list(APPEND CLIENT_SRC ${PORT_SRC})
+
+if(OC_SECURITY_ENABLED)
+    file(GLOB SECURITY_SRC
+        ${PROJECT_SOURCE_DIR}/security/oc_*.c
+    )
+    list(REMOVE_ITEM SECURITY_SRC
+        ${PROJECT_SOURCE_DIR}/security/oc_obt*.c
```

Comment:
I don't think `list` operations support globbing. You can first `file(GLOB)` the obt files and then remove them.

Reply:
You're right. Fixed in https://github.com/iotivity/iotivity-lite/pull/181/commits/6cafe455fddb893be043fa64cd3acdf873fda661